operations/dataGenV2.go

```diff
@@ -0,0 +1,332 @@
+	}
+	Log.Debug.Printf("Done %s", userId)
+}
+
+func FormatOutput(segmentConfig config.UserSegmentV2, userId string, event string, eventCounter int, userAttributes map[string]string, eventAttributes map[string]string) (string){
+
+	type output struct {
+		UserId string `json:"user_id"`
+		Event string `json:"event_name"`
+		Timestamp int `json:"timestamp"`
+		UserAttributes map[string]string `json:"user_properties"`
+		EventAttributes map[string]string `json:"event_properties"`
+	}
+
+	var op output
+	op.UserId = userId
+	op.Event = event
+	op.Timestamp, _ = strconv.Atoi(fmt.Sprintf("%v", segmentConfig.Start_Time.Add(time.Second * time.Duration(eventCounter * segmentConfig.Activity_ticker_in_seconds)).Unix()))
+	op.UserAttributes = userAttributes
+	op.EventAttributes = eventAttributes
+	e, _ := json.Marshal(&op)
+	return string(e)
+}
+
+func WaitIfRealTime(duration int) {
+	if(IsRealTime() == true){
+		time.Sleep(time.Duration(duration) * time.Second)
+	}
+}
+
+func IsRealTime() bool {
+	if(config.ConfigV2.Real_Time == true){
+		return true
+	}
+	return false
+}
+func SetUserAttributes(segmentConfig config.UserSegmentV2, userId string) map[string]string{
+	var userAttributes map[string]string
+	if(segmentConfig.Set_attributes == true){
+		attr := segmentConfig.User_attributes[userId]
+		if(attr != nil){
+			userAttributes = attr
+		}
+	}
+	return userAttributes
+}
+
+func SetEventAttributes(segmentConfig config.UserSegmentV2,eventName string) map[string]string{
+	var eventAttributes map[string]string
+	if(segmentConfig.Set_attributes == true){
+		attr := segmentConfig.Event_attributes[eventName]
+		if(attr != nil){
+			eventAttributes = attr
+		}
+	}
+	return eventAttributes
+}
+
+func GetRandomActivity(probMap SegmentProbMap) string {
+	activity := rand.Intn(probMap.activityMultiplier)
+	activityName, state := probMap.activityProbRangeMap.Get(activity)
+	if(state == false){
+		Log.Error.Fatal(fmt.Sprintf("ActivityProbablityRangeMap: Key not found %v", activity))
+	}
+	return activityName
+}
+
+func GetRandomEvent(probMap SegmentProbMap) string {
+	event := rand.Intn(probMap.eventMultiplier)
+	eventName, state := probMap.eventProbRangeMap.Get(event)
+	if(state == false){
+		Log.Error.Fatal(fmt.Sprintf("EventProbablityRangeMap: Key not found %v", event))
+	}
+	return eventName
+}
+
+func GetRandomEventWithCorrelation(lastKnownGoodState *string, seedEvents []string, probMap SegmentProbMap) (string) {
+	if *lastKnownGoodState == "" {
+		*lastKnownGoodState = seedEvents[rand.Intn(len(seedEvents))]
+		return *lastKnownGoodState
+	}
+    
+	event := rand.Intn(probMap.EventCorrelationMultiplier[*lastKnownGoodState])
+	eventName, state := probMap.EventCorrelationRangeMap[*lastKnownGoodState].Get(event)
+	if(state == false){
+		Log.Error.Fatal(fmt.Sprintf("EventProbablityRangeMapWithCorrelation: Key not found %v", event))
+	}
+	*lastKnownGoodState = eventName
+	return eventName
+}
+
+func ComputeRangeMap(probMap map[string]float64) (utils.RangeMap, int) {
```

Comment:
Further we should add unit tests for all important methods.

Reply:
Done